hamming: update README, test, stub, example

hanjiexi opened the pull request:
This is my first time submitting to exercism, please let me know if there is anything wrong with my commits or the pull request and I will fix it.

snahor commented on exercises/hamming/example.sml:
```diff
@@ -1,10 +1,12 @@
-exception NonEqualLengthStringsFound;
```
I think we should keep this and use it instead of `Fail "..."`.

snahor commented:
Thanks @hanjiexi and welcome!

hanjiexi commented:
I changed it to the `Fail` from the exception because the new test generated by `bin/generate` got rid of the `NonEqualLengthStringsFound` exception and replaced it with `Fail "left and right strands must be of equal length"`. Am I supposed to change something in the test generator (which I don't know how to do) or is it fine to make manual edits to a generated test?

mcmillhj commented:
@hanjiexi It is fine to make manual updates after the generator has run. The generator pulls the test cases from the problem-specifications repo's canonical-data.json file. In some cases there are better solutions (e.g. custom error type) that are worth editing for.

hanjiexi commented:
Okay thanks for the feedback, I've just pushed the requested update.
